src/index.js

```diff
@@ -670,7 +671,18 @@ const members = {
       return this.getAttribute('name');
     },
     set(name) {
-      return this.setAttribute('name', name);
+      const ret = this.__setAttribute('name', name);
+      if (!isSlotNode(this)) {
+        return ret;
+      }
+      const root = slotToRootMap.get(this);
+      const slotHasRoot = root;
+
+      if (slotHasRoot) {
+        removeSlotFromRoot(root, this);
```

Comment:
I think it's because the previous DOM nodes must be unassigned. There might be a better way to do this, like `unassignNodesFromSlot()`. I'm thinking this might be able to be used elsewhere, too.

That said, I'd get this working and tests green before trying to refactor that now.